src/Symfony/Component/Form/Form.php

```diff
@@ -344,7 +344,7 @@ public function setData($modelData)
         }
 
         // Treat data as strings unless a value transformer exists
-        if (!$this->config->getViewTransformers() && !$this->config->getModelTransformers() && is_scalar($modelData)) {
+        if (!$this->config->getViewTransformers() && !$this->config->getModelTransformers() && !is_bool($modelData) && is_scalar($modelData)) {
```

Comment:
You should replace the scalar check by a string check instead of testing a non-bool.